### PR TITLE
Feature/autodiff shape manipulations

### PR DESCRIPTION
**Pull request by VIVelev**
Add Reshape and Transform to the set of autodiff nujo functions (used in ConvNets)

### CODE REVIEW

**Review by VIVelev — COMMENTED** [review]
LGTM

File: tmp.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import nujo as nj
```
Comment (ManiacMaxo) [inline tmp.py:1]: I don't think this file should be here

File: nujo/autodiff/_functions/_transform.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+        super(_Reshape, self).__init__(input, name=self.__class__.__name__)
+        self.shape = shape
+        self._input_shape = self.children[0].shape
+
+    def forward(self) -> ndarray:
+        return self.children[0].value.reshape(self.shape)
+
+    def backward(self) -> Tuple[ndarray]:
+        return self._input_shape,
```
Comment (ManiacMaxo) [inline nujo/autodiff/_functions/_transform.py:29]: comma

File: nujo/autodiff/_functions/_transform.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        self.dims = dims if dims is not None else reversed(
+            range(len(self.dims)))
+        self._detranspose_dims = sorted(range(len(self.dims)),
+                                        key=lambda idx: self.dims[idx])
+
+    def forward(self) -> ndarray:
+        return self.children[0].value.transpose(*self.dims)
+
+    def backward(self) -> Tuple[ndarray]:
+        return self._detranspose_dims,
```
Comment (ManiacMaxo) [inline nujo/autodiff/_functions/_transform.py:50]: same

File: tmp.py (hunk 1 above)
Comment (VIVelev) [reply tmp.py:1]: Will get remove, but in later PR, its needed for now

File: nujo/autodiff/_functions/_transform.py (hunk 2 above)
Comment (VIVelev) [reply nujo/autodiff/_functions/_transform.py:29]: The func needs to return a Tuple, so that's how it should be

**Comment by VIVelev** [comment]
Will merge this one, a refresh PR is on the way.

**Pull request merged**